framework/components/ADataTable/ADataTable.js

```diff
@@ -1,14 +1,22 @@
 import PropTypes from "prop-types";
-import React, {forwardRef} from "react";
+import React, {forwardRef, useState} from "react";
 
 import AIcon from "../AIcon";
 import ASimpleTable from "../ASimpleTable";
 import "./ADataTable.scss";
 
-const ADataTable = forwardRef(
-  ({className: propsClassName, headers, items, onSort, sort, ...rest}, ref) => {
-    let className = "a-data-table";
+const TableHeader = (props) => <th role='columnheader' className='a-data-table__header' {...props} />;
+const TableRow = (props) => <tr role='row' className='a-data-table__row' {...props} />;
+const TableCell = (props) => <td role='cell' className='a-data-table__cell' {...props} />;
 
+const ADataTable = forwardRef(
+  ({className: propsClassName, expandable, headers, items, onSort, sort, ...rest}, ref) => {
+    const [expandedRows, setExpandedRows] = useState({});
+    const ExpandableComponent = expandable?.component;
```

Comment:
This might be worth wrapping in `useMemo` so ExpandableComponent isn't redefined every time some other prop changes.

```suggestion
    const ExpandableComponent = useMemo(() => expandable?.component, [expandable]);
```

You could also `useMemo` again around the sort logic starting on line 24 (unchanged in this PR).